server/genquery2/src/genquery2_sql.cpp

```diff
@@ -0,0 +1,1172 @@
+            if (table_names.at(i) == _table_name) {
+                return i;
+            }
+        }
+
+        throw std::invalid_argument{fmt::format("table [{}] not supported", _table_name)};
+    } // to_index
+
+    // clang-format off
+    constexpr auto table_edges = std::to_array<edge_type>({
+        {to_index("R_COLL_MAIN"), to_index("R_DATA_MAIN")},                 // R_COLL_MAIN.coll_id = R_DATA_MAIN.coll_id
+        {to_index("R_COLL_MAIN"), to_index("R_OBJT_ACCESS")},               // R_COLL_MAIN.coll_id = R_OBJT_ACCESS.object_id
+        {to_index("R_COLL_MAIN"), to_index("R_OBJT_METAMAP")},              // R_COLL_MAIN.coll_id = R_OBJT_METAMAP.object_id
+        {to_index("R_COLL_MAIN"), to_index("R_TICKET_MAIN")},               // R_COLL_MAIN.coll_id = R_TICKET_MAIN.object_id
+
+        {to_index("R_DATA_MAIN"), to_index("R_OBJT_ACCESS")},               // R_DATA_MAIN.data_id = R_OBJT_ACCESS.object_id
+        {to_index("R_DATA_MAIN"), to_index("R_OBJT_METAMAP")},              // R_DATA_MAIN.data_id = R_OBJT_METAMAP.object_id
+        {to_index("R_DATA_MAIN"), to_index("R_RESC_MAIN")},                 // R_DATA_MAIN.resc_id = R_RESC_MAIN.resc_id
+        {to_index("R_DATA_MAIN"), to_index("R_TICKET_MAIN")},               // R_DATA_MAIN.data_id = R_TICKET_MAIN.object_id
+
+        {to_index("R_META_MAIN"), to_index("R_OBJT_METAMAP")},              // R_META_MAIN.meta_id = R_OBJT_METAMAP.meta_id
+
+        {to_index("R_OBJT_ACCESS"), to_index("R_TOKN_MAIN")},               // R_OBJT_ACCESS.access_type_id = R_TOKN_MAIN.token_id
+
+        {to_index("R_OBJT_METAMAP"), to_index("R_RESC_MAIN")},              // R_OBJT_METAMAP.object_id = R_RESC_MAIN.resc_id
+        {to_index("R_OBJT_METAMAP"), to_index("R_USER_MAIN")},              // R_OBJT_METAMAP.object_id = R_USER_MAIN.user_id
+
+        {to_index("R_TICKET_MAIN"), to_index("R_USER_MAIN")},               // R_TICKET_MAIN.user_id = R_USER_MAIN.user_id
+
+        {to_index("R_TICKET_MAIN"), to_index("R_TICKET_ALLOWED_HOSTS")},    // R_TICKET_MAIN.ticket_id = R_TICKET_ALLOWED_HOSTS.ticket_id
+        {to_index("R_TICKET_MAIN"), to_index("R_TICKET_ALLOWED_USERS")},    // R_TICKET_MAIN.ticket_id = R_TICKET_ALLOWED_USERS.ticket_id
+        {to_index("R_TICKET_MAIN"), to_index("R_TICKET_ALLOWED_GROUPS")},   // R_TICKET_MAIN.ticket_id = R_TICKET_ALLOWED_GROUPS.ticket_id
+
+        {to_index("R_USER_MAIN"), to_index("R_USER_AUTH")},                 // R_USER_MAIN.user_id = R_USER_AUTH.user_id
+        {to_index("R_USER_MAIN"), to_index("R_USER_GROUP")},                // R_USER_MAIN.user_id = R_USER_GROUP.group_user_id
+        {to_index("R_USER_MAIN"), to_index("R_USER_PASSWORD")},             // R_USER_MAIN.user_id = R_USER_PASSWORD.user_id
+        {to_index("R_USER_MAIN"), to_index("R_USER_SESSION_KEY")},          // R_USER_MAIN.user_id = R_USER_SESSION_KEY.user_id
+
+        // Handle R_USER_GROUP?
+        // Handle R_QUOTA_MAIN
+        // Handle R_QUOTA_USAGE
```

Comment:
Got any issue(s) to link for these?

Reply:
Going to capture all of these in comments once we all agree the code changes are good.